tools/run_parallel.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Uncommenting the next line may be useful when debugging this script.
+#set -x
+
+usage () {
+    cat << EOF
+Usage: run_parallel.sh {command} {num_iter} {num_parallel}
+Where:
+  {command} is a string containing the command, including parameters, to run
+  {num_iter} is an positive integer indicating how many iterations to execute
+  {num_parallel} is an (optional) positive integer indicating how many parallel commands should be executed
+     in each iteration. If not provided, the default is half the number of available CPU cores.
+EOF
+}
+
+if [ "$#" -lt  2 ]; then
+    echo "Illegal number of parameters."
+    usage
+    exit 1
+fi
+
+# Calculate the number of CPU cores. This code is Linux specific at the moment.
+NCORES=$(echo "`grep -c ^processor /proc/cpuinfo` / 2" | bc)
+
+command=$1
+num_iter=$2
+
+if [ "$#" -eq  3 ]; then
+  num_parallel=$3
+else
+  num_parallel=$NCORES
+fi
+
+echo "run_parallel:"
+echo "  number of cores: $NCORES"
+echo "  command:         $command"
+echo "  num_parallel:    $num_parallel"
+echo "  num_iter:        $num_iter"
+
+outf=./outfile.txt
+
+for i in $(seq 1 $num_iter); do
+  echo "Starting iteration $i" >> $outf
```

Comment:
Do you want to clear the contents of `$outf` at some stage so that the script can be called multiple times without obsolete contents accumulated? Same for `nohup.out.$t`

Reply:
@lukech , that would be a useful option if were to use the script multiple times in sequence and not want to accumulate the output from multiple runs. I suggest we add it later, if required.